delphi-frontend/src/main/java/au/com/integradev/delphi/cfg/ControlFlowGraphImpl.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2025 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.cfg;
+
+import au.com.integradev.delphi.cfg.api.Block;
+import au.com.integradev.delphi.cfg.api.ControlFlowGraph;
+import au.com.integradev.delphi.cfg.api.Terminated;
+import au.com.integradev.delphi.cfg.api.UnconditionalJump;
+import au.com.integradev.delphi.cfg.block.BlockImpl;
+import com.google.common.collect.Lists;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class ControlFlowGraphImpl implements ControlFlowGraph {
+  private Block entry;
+  private final Block exit;
+  private final List<Block> blocks;
+
+  public ControlFlowGraphImpl(Block entry, Block exit, List<Block> blocks) {
+    this.entry = entry;
+    this.exit = exit;
+    this.blocks = blocks;
+  }
+
+  @Override
+  public Block getEntryBlock() {
+    return entry;
+  }
+
+  @Override
+  public Block getExitBlock() {
+    return exit;
+  }
+
+  @Override
+  public List<Block> getBlocks() {
+    return Collections.unmodifiableList(Lists.reverse(blocks));
+  }
+
+  /// Removes redundant blocks from the graph and updates their neighbouring blocks.
+  public ControlFlowGraphImpl pruned() {
```

Comment:
`ControlFlowGraphImpl::pruned` returning a `ControlFlowGraphImpl` makes it seem like the graph is cloned, when it's actually pruned inplace. It might be better to remove the return type and rename to `prune()` for clarity.